docs/content/sips/000-sqlite.md

```diff
@@ -0,0 +1,139 @@
+
+* For the 20% of use cases where `wasi-sql` is too generic, a dedicated `sqlite` interface can provide that functionality. 
+* The `wasi-sql` spec is under active investigation, and there are large questions about how to best support such a wide breadth of sql flavors. This implementation can help clarify those questions and push upstream work further along.
+
+## Proposal
+
+In order to support sqlite, the following need to be added to Spin:
+
+- A `WIT` file that defines the sqlite interface
+- SDK implementations for various programming languages
+- A default local sqlite store (note: Spin already uses sqlite for the KV implementation so this should be trivial)
+- Potentially runtime configuration for configuring how sqlite is provisioned.
+- Potentially a mechansim for handling database migrations
+
+### Interface (`.wit`)
+
+We will start with the `wasi-sql` interface but deliberately change that interface as to better match sqlite semantics. This will ensure that we're not simply implementing early versions of the `wasi-sql` interface while still having good answers for why the interface differs when it does.
+
+Like `wasi-sql` and the key-value store, we model resources such as database connections as pseudo-[resource handles](https://github.com/WebAssembly/component-model/blob/main/design/mvp/WIT.md#item-resource) which may be created using an `open` function and disposed using a `close` function. Each operation on a connection is a function which accepts a handle as its first parameter.
+
+Note that the syntax of the following `WIT` file matches the `wit-bindgen` version currently used by Spin, which is out-of-date with respect to the latest [`WIT` specification](https://github.com/WebAssembly/component-model/blob/main/design/mvp/WIT.md) and implementation. Once we're able to update `wit-bindgen`, we'll update the syntax of all the Spin `WIT` files, including this one.
+
+```fsharp
+// A handle to an open sqlite instance
+type connection = u32
+
+// The set of errors which may be raised by functions in this interface
+variant error {
+  // The host does not recognize the database name requested.
+  no-such-database,
+  // Some implementation-specific error has occurred (e.g. I/O)
+  io(string)
+}
+
+// Open a connection to a named database instance.
+//
+// If `database` is "default", the default instance is opened.
+//
+// `error::no-such-database` will be raised if the `name` is not recognized.
+open: func(name: string) -> expected<connection, error>
+
+// TODO: Do we want a separate statment type? This is currently what `wasi-sql` does.
+// allows parameterized queries
+type statement = u32
+drop-statement: func(statement: statement)
+prepare-statement: func(query: string, params: list<data-type>) -> expected<statement, error>
```

Comment:
My impression of a prepared statement was that it generally didn't include parameter values (or, at least, that you could reuse it with different parameter values).  This interface seems to bake the parameter values into the prepared statement.  Is that intentional?